`other.test_split_module` regression

```
C:\emsdk\emscripten\main>test\runner other.test_split_module
Running 1 tests
[1/1] test_split_module (test_other.other.test_split_module) ... FAIL

Stdout:
-- begin program output --
C:\emsdk\emscripten\main\out\test\test_split_module.js:652
  var instance = new WebAssembly.Instance(module, info);
                 ^

LinkError: WebAssembly.Instance(): table import 9 has a larger maximum size 14 than the module's declared maximum 8
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at test_split_module.wasm.trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:wasm-function[30]:0x17b0)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_split_module.js:725:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_split_module.js:742:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_split_module.js:768:17)
Thrown at:
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at $trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:1:6065)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)


Node.js v22.16.0
-- end program output --


======================================================================
FAIL: test_split_module (test_other.other.test_split_module)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\emsdk\emscripten\main\test\common.py", line 380, in resulting_test
    return func(self, *args)
  File "C:\emsdk\emscripten\main\test\decorators.py", line 419, in metafunc
    return func(self, *args, **kwargs)
  File "C:\emsdk\emscripten\main\test\test_other.py", line 12476, in test_split_module
    result = self.run_js('test_split_module.js')
  File "C:\emsdk\emscripten\main\test\common.py", line 1025, in run_js
    self.fail('JS subprocess failed (%s): %s (expected=%s).  Output:\n%s' % (error.cmd, error.returncode, assert_returncode, ret))
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: JS subprocess failed (C:/emsdk/node/22.16.0_64bit/bin/node.exe --stack-trace-limit=50 --trace-uncaught C:\emsdk\emscripten\main\out\test\test_split_module.js): 1 (expected=0).  Output:
C:\emsdk\emscripten\main\out\test\test_split_module.js:652
  var instance = new WebAssembly.Instance(module, info);
                 ^

LinkError: WebAssembly.Instance(): table import 9 has a larger maximum size 14 than the module's declared maximum 8
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at test_split_module.wasm.trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:wasm-function[30]:0x17b0)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_split_module.js:725:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_split_module.js:742:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_split_module.js:768:17)
Thrown at:
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at $trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:1:6065)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)


Node.js v22.16.0


Stdout:
-- begin program output --
C:\emsdk\emscripten\main\out\test\test_split_module.js:652
  var instance = new WebAssembly.Instance(module, info);
                 ^

LinkError: WebAssembly.Instance(): table import 9 has a larger maximum size 14 than the module's declared maximum 8
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at test_split_module.wasm.trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:wasm-function[30]:0x17b0)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)
    at receiveInstance (C:\emsdk\emscripten\main\out\test\test_split_module.js:725:5)
    at receiveInstantiationResult (C:\emsdk\emscripten\main\out\test\test_split_module.js:742:12)
    at createWasm (C:\emsdk\emscripten\main\out\test\test_split_module.js:768:17)
Thrown at:
    at instantiateSync (C:\emsdk\emscripten\main\out\test\test_split_module.js:652:18)
    at ret (C:\emsdk\emscripten\main\out\test\test_split_module.js:637:13)
    at $trampoline_say_hello (wasm://wasm/test_split_module.wasm-0007ecf6:1:6065)
    at C:\emsdk\emscripten\main\out\test\test_split_module.js:583:12
    at saveProfileData (C:\emsdk\emscripten\main\out\test\test_split_module.js:1764:16)
    at callRuntimeCallbacks (C:\emsdk\emscripten\main\out\test\test_split_module.js:818:26)
    at postRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:520:3)
    at doRun (C:\emsdk\emscripten\main\out\test\test_split_module.js:1682:5)
    at run (C:\emsdk\emscripten\main\out\test\test_split_module.js:1693:5)
    at removeRunDependency (C:\emsdk\emscripten\main\out\test\test_split_module.js:852:11)


Node.js v22.16.0
-- end program output --

----------------------------------------------------------------------
Ran 1 test in 1.174s

FAILED (failures=1)

```

Similar to #26936, this test is failing, but only after updating to latest LLVM. Reproes on my CI on Linux, Windows and Mac.

Started about a day ago, see e.g. http://clbri.com:8010/#/builders/11

--- a/src/ir/module-splitting.cpp
+++ b/src/ir/module-splitting.cpp
@@ -1107,7 +1107,11 @@ void ModuleSplitter::setupTablePatching() {
     // so we should export and import the active table here.
     auto secondaryTable =
       secondary.getTableOrNull(tableManager.activeTable->name);
-    if (!secondaryTable) {
+    if (secondaryTable) {
+      // In case it's already in the secondary module, sync the initial/max
+      secondaryTable->initial = tableManager.activeTable->initial;
+      secondaryTable->max = tableManager.activeTable->max;
+    } else {
       secondaryTable =
         ModuleUtils::copyTable(tableManager.activeTable, secondary);
       makeImportExport(*tableManager.activeTable,

--- a/test/lit/wasm-split/active-table-base-global-used-elsewhere.wast
+++ b/test/lit/wasm-split/active-table-base-global-used-elsewhere.wast
@@ -18,8 +18,8 @@
 
  ;; PRIMARY:      (import "placeholder.deferred" "1" (func $placeholder_1))
 
- ;; PRIMARY:      (table $table 2 funcref)
- (table $table 1 funcref)
+ ;; PRIMARY:      (table $table 2 2 funcref)
+ (table $table 1 1 funcref)
  (elem (global.get $base) $keep)
  ;; PRIMARY:      (elem $0 (global.get $base) $keep $placeholder_1)
 
@@ -40,7 +40,7 @@
  (func $keep
   (call $split)
  )
- ;; SECONDARY:      (import "primary" "table" (table $table 1 funcref))
+ ;; SECONDARY:      (import "primary" "table" (table $table 2 2 funcref))
 
  ;; SECONDARY:      (import "primary" "global" (global $base i32))
 

